tests/test_api.py

```diff
@@ -167,58 +167,263 @@ class TestApi(unittest.TestCase):
+                'status': 'enabled',
+                'response': {
+                    'code': 200
+                }
+            },
+            {
+                'name': "test_put_user1@email.com",
+                'status': 'disabled',
+                'response': {
+                    'code': 200
+                }
+            }
+        ]
+        for test in tests:
+            with self.subTest():
+                headers = {'Content-Type': "application/json"}
+                headers.update(get_auth_header(service_accounts['admin']))
+                url = furl(f'/v1/users/{test["name"]}')
+                query_params = {
+                    'user_id': test['name'],
+                    'status': test['status']
+                }
+                url.add(query_params=query_params)
+                user = User.provision_user(directory, test['name'])
+                if test['status'] == 'disabled':
+                    user.enable()
+                resp = self.app.put(url.url, headers=headers)
+                self.assertEqual(test['response']['code'], resp.status_code)
+                resp.raise_for_status()
 
-    @unittest.skip("incomplete")
-    def test_put_group(self):
-        pass
+    def test_put_users_groups(self):
+        tests = [
+            {
+                'name': "test_put_user_group0@email.com",
+                'action': 'add',
+                'json_request_body': {
+                    "groups": [Group.create(directory, "group_0").name]
+                },
+                'response': {
+                    'code': 200
+                }
+            },
+            {
+                'name': "test_put_user_group1@email.com",
+                'action': 'remove',
+                'json_request_body': {
+                    "groups": [Group.create(directory, "group_1").name]
+                },
+                'response': {
+                    'code': 200
+                }
+            }
+        ]
+        for test in tests:
+            with self.subTest(test['json_request_body']):
+                data = json.dumps(test['json_request_body'])
+                headers = {'Content-Type': "application/json"}
+                headers.update(get_auth_header(service_accounts['admin']))
+                url = furl(f'/v1/users/{test["name"]}/groups/')
+                query_params = {
+                    'user_id': test['name'],
+                    'action': test['action']
+                }
+                url.add(query_params=query_params)
+                user = User.provision_user(directory, test['name'])
+                if test['action']=='remove':
+                    user.add_groups(test['json_request_body']['groups'])
+                resp = self.app.put(url.url, headers=headers, data=data)
+                self.assertEqual(test['response']['code'], resp.status_code)
+                resp.raise_for_status()
+
+    def test_get_users_groups(self):
```

Comment:
change to test_get_username_groups

Reply:
add a test where the user is in a group or two and groups are returned from the API